src/components/modals/CompleteQuestModal.tsx

```diff
@@ -138,60 +141,257 @@ const RedeemItem = ({ questName }: Omit<QuestDetailProps, "type">) => {
+        if (provingStateUpdate.numRequirementsUpdate) {
+          newProvingState.numRequirementsTotal =
+            provingStateUpdate.numRequirementsUpdate.numRequirementsTotal;
+          newProvingState.numRequirementsProven =
+            provingStateUpdate.numRequirementsUpdate.numRequirementsProven;
+        }
+        if (provingStateUpdate.currentRequirementUpdate) {
+          newProvingState.currentRequirementNumSigsTotal =
+            provingStateUpdate.currentRequirementUpdate.currentRequirementNumSigsTotal;
+          newProvingState.currentRequirementNumSigsProven =
+            provingStateUpdate.currentRequirementUpdate.currentRequirementNumSigsProven;
+        }
+
+        return newProvingState;
+      });
+    };
+
+    const serializedProof = await generateProofForQuest(
+      quest,
+      onUpdateProvingState
+    );
+
+    const response = await fetch("/api/quest/submit_proof", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        questId: quest.id.toString(),
+        authToken: authToken.value,
+        serializedProof,
+      }),
+    });
+
+    if (!response.ok) {
+      toast.error("Failed to submit proof!");
+      setDisplayState(CompleteQuestDisplayState.INITIAL);
+      return;
+    }
+
+    const data = await response.json();
+    if (!data.verified) {
+      toast.error("Proof failed to verify!");
+      setDisplayState(CompleteQuestDisplayState.INITIAL);
+      return;
+    }
+
+    const proofId = data.proofId;
+    if (!proofId) {
+      toast.error("Failed to submit proof!");
+      setDisplayState(CompleteQuestDisplayState.INITIAL);
+      return;
+    }
+
+    const senderPrivateKey = keys.encryptionPrivateKey;
+    const recipientPublicKey = profile.encryptionPublicKey;
+    const encryptedMessage = await encryptQuestCompletedMessage({
+      questId: quest.id.toString(),
+      questName: quest.name,
+      proofId,
+      senderPrivateKey,
+      recipientPublicKey,
+    });
+
+    // Send quest completed info as encrypted jubSignal message to self
+    try {
+      const response = await fetch("/api/messages", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          message: encryptedMessage,
+          recipientPublicKey,
+          token: authToken.value,
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error("Error sending message");
+      }
+    } catch (error) {
+      console.error(
+        "Error sending encrypted quest completed info to server: ",
+        error
+      );
+      toast.error(
+        "An error occured while completing the quest. Please try again."
+      );
+      setDisplayState(CompleteQuestDisplayState.INITIAL);
+      return;
+    }
+
+    // Update activity feed in local storage
+    try {
+      await loadMessages({ forceRefresh: false });
```

Comment:
Do we need to load all the messages again? The only change in activity feed will be the completed quest right?